.github/workflows/ci-rerun.yml

```
@@ -0,0 +1,105 @@
+name: Re-run Workflow
+
+on:
+  workflow_run:
+    workflows:
+      - CI
+    types:
+      - completed
+  pull_request_target:
+    types:
+      - closed
+      - labeled
+      - unlabeled
+  schedule:
+    - cron: '30 */3 * * *' # every 3 hours (30 minutes past the hour)
+  workflow_dispatch:
+
+env:
+  GH_REPO: ${{ github.repository }}
+
+permissions:
+  contents: read
+  actions: write
+
+jobs:
+  scheduled-retry:
+    if: >
+      github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'
```

Might be cleaner to do something like this:
```suggestion
    if: contains(fromJSON('["schedule", "workflow_dispatch"]'), github.event.name)
```
See https://docs.github.com/en/actions/learn-github-actions/expressions#example-matching-an-array-of-strings